### DISCUSSION TITLE
Index is too large

### DISCUSSION THREAD

**Discussion by wanxings** [zinclabs/zinc#437] [Q&A] [answered]
I have 20 million articles, I tried to index 80 thousand article titles, the index size is 215.59 MB, the indexed data looks like the following。
`{
    "name": "[How to implement a stack in Golang - Educativeio](in-golang)",
    "hash": "55418169b131817de53a485a8",
    "create_time": 1581149329,
    "user": 2,
    "length": 995588994,
    "level": 0,
    "style": 0,
    "update_time": 1581149329,
    "weights": 2
}`
Mappings:
![image](https://user-images.githubusercontent.com/37164409/184608469-8ccb5034-6db4-4767-89db-848d767d19db.png)

If all indexes will reach nearly 60GB, is this normal? It's only 5GB in my mysql,Is it the Mappings setting or something else?
Thanks🙏

**Comment by prabhatsharma** [comment, marked as answer]
We us compression to reduce size of the indexes. We also use sharding to improve performance. We use 2 layers of sharding. The more shards you have, the higher the size will be. You can try reducing the size of the index by tuning the following environment variables.

ZINC_SHARD_NUM=1

// For 1 GB default shard size. You can change it to 10 GB to reduce index size.
ZINC_SHARD_MAX_SIZE=1073741824
